test/compilable/emptygenmain.d

```diff
@@ -0,0 +1,3 @@
+// REQUIRED_ARGS: -main -c
```

Comment:
I would have liked a link to a bugzilla issue (or this PR since there is none) to make it clear what this test is doing.

Reply:
This issue was introduced in an unrealesed version (master) so I don't see the need of a bugzilla issue for unreleased regressions. Anyway, if we do it, I can create one.

Reply:
Also, if someone want to know the rationale of this test, they can do `git blame test/compilable/emptygenmain.d` and read the commit body, since I added a brief description of the problem there.